pkg/distributor/influx.go

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributor
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/grpcutil"
+	"github.com/grafana/dskit/httpgrpc"
+	"github.com/grafana/dskit/middleware"
+	io2 "github.com/influxdata/influxdb/v2/kit/io"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/mimir/pkg/distributor/influxpush"
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/util"
+	utillog "github.com/grafana/mimir/pkg/util/log"
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+func parser(ctx context.Context, r *http.Request, maxSize int, _ *util.RequestBuffers, req *mimirpb.PreallocWriteRequest, logger log.Logger) error {
+	spanLogger, ctx := spanlogger.NewWithLogger(ctx, logger, "Distributor.InfluxHandler.decodeAndConvert")
+	defer spanLogger.Span.Finish()
+
+	spanLogger.SetTag("content_type", r.Header.Get("Content-Type"))
+	spanLogger.SetTag("content_encoding", r.Header.Get("Content-Encoding"))
+	spanLogger.SetTag("content_length", r.ContentLength)
+
+	ts, bytesRead, err := influxpush.ParseInfluxLineReader(ctx, r, maxSize)
+	// TODO(alexg): one argument for splitting up the decoding and conversion is to facilitate granular timings
+	// right now since ParseInfluxLineReader() does both
+	// The otel version decodes the whole input and then processes it, the existing Influx code parses each line as it
+	// decodes it.
+	level.Debug(spanLogger).Log(
+		"msg", "decodeAndConvert complete",
+		"bytesRead", bytesRead,
+	)
+	if err != nil {
+		level.Error(logger).Log("err", err.Error())
```

Comment:
Does this logger have all the required context? Can you also add some context here about what was going on when this happened? I'm scared of finding this log:
```
ts=2024-12-17 err="unexpected EOF"
```

Also, nit, `.Error()` call is not needed, just pass `err`.

Reply:
Agreed, fixed in next push.